src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -264,6 +264,8 @@ public void workspaceInit(
     blockWaitingModule =
         Preconditions.checkNotNull(runtime.getBlazeModule(BlockWaitingModule.class));
     credentialModule = Preconditions.checkNotNull(runtime.getBlazeModule(CredentialModule.class));
+    repositoryModule =
```

Comment:
This broken the `BuildWithoutTheBytesIntegrationTest` internally with
```
java.lang.NullPointerException
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:966)
	at com.google.devtools.build.lib.remote.RemoteModule.workspaceInit(RemoteModule.java:268)
	at com.google.devtools.build.lib.runtime.BlazeRuntime.initWorkspace(BlazeRuntime.java:256)
	at com.google.devtools.build.lib.buildtool.util.BlazeRuntimeWrapper.<init>(BlazeRuntimeWrapper.java:150)
	at com.google.devtools.build.lib.buildtool.util.BuildIntegrationTestCase$2.<init>(BuildIntegrationTestCase.java:360)
	at com.google.devtools.build.lib.buildtool.util.BuildIntegrationTestCase.createRuntimeWrapper(BuildIntegrationTestCase.java:360)
	at com.google.devtools.build.lib.buildtool.util.BuildIntegrationTestCase.createFilesAndMocks(BuildIntegrationTestCase.java:281)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:61)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:308)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:368)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:333)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:81)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:295)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:308)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:420)
	at com.google.testing.junit.runner.internal.junit4.CancellableRequestFactory$CancellableRunner.run(CancellableRequestFactory.java:108)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:141)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:119)
	at com.google.testing.junit.runner.junit4.JUnit4Runner.run(JUnit4Runner.java:100)
	at com.google.testing.junit.runner.RunnerShell$2.run(RunnerShell.java:34)
	at com.google.testing.junit.runner.GoogleTestRunner.runTestsInSuite(GoogleTestRunner.java:179)
	at com.google.testing.junit.runner.GoogleTestRunner.runTestsInSuite(GoogleTestRunner.java:162)
	at com.google.testing.junit.runner.GoogleTestRunner.main(GoogleTestRunner.java:122)
```

Moving this https://cs.opensource.google/bazel/bazel/+/master:src/test/java/com/google/devtools/build/lib/buildtool/util/BuildIntegrationTestCase.java;l=649-650 out of the if statement seems to fix it, but I'm not sure if it'll cause any other problems, we can try.

Also, is there a better way to avoid referencing BazelRepositoryModule from RemoteModule?

@criemen @fmeum

Reply:
We were just discussing another (further) refactoring of moving `HttpDownloader` (and maybe the download manager) into `BlazeRuntime` in https://github.com/bazelbuild/bazel/pull/23106#discussion_r1691318199. I'l ltry that, that'd also remove the need to pull BazelRepositoryModule into RemoteModule, and setting it up for all the tests.

Reply:
SG, thanks!

Reply:
or the `CommandEnvironment`, if we can get away with that, actually